### ISSUE TITLE
Android only: ability to configure VOLUME_UP/VOLUME_DOWN device physical keys as shortcuts for actions

### ISSUE THREAD

**Issue by StLarry7** [androidseb/mapmarker#40]
## Developer specifications

Add two settings allowing to set either the VOLUME_UP and/or the VOLUME_DOWN device physical keys as a shortcut to immediately perform one of these actions:

* add a marker based on where the screen center aims at on the map
* add a marker based on where the current device location is
* start recording a new track

Notes about this feature:

* both settings are OFF by default, the user must go and activate them explicitly, since it will prevent the user from changing the volume when this setting is active with the app in the foreground
* this only works when the device screen is on and Map Marker is the app currently open
* this is only for Android: iOS limitations do not allow this to happen, and web is meant to be used from computers

## Initial description by user

I use Map Marker for mapping environmental features using Android, and often have to record hundreds of waypoints while also navigating around obstacles.  I use the POIs to create features and maps in GIS back in the office.  
I'd like to program a bluetooth button to record POI's, rather than tapping on the touch screen, which can be awkward while in the field.   

Would it be possible to add a remote control option for adding POI's?   There are dozens of cheap bluetooth camera shutter or phonecall remote control buttons available.  I could use a macro/Tasker app to program the workflow. similar to this:  https://xdaforums.com/t/diy-tasker-use-any-bluetooth-headset-as-remote-camera-shutter.2769359/

But I'd like to have colleagues collect data as well, and if the remote control functionality was integrated into the app it would make life incredibly easier.

**Comment by androidseb** [comment]
Hello, thanks for suggesting this idea! Do you really need to record specific points, or would [recording a track](https://github.com/androidseb/mapmarker/issues/4) (as a line-path) be more practical?

**Comment by fieldcam** [comment]
Recording a track also a good feature.

**Comment by StLarry7** [comment]
Yeah.   I wish it were that easy, as that's the obvious solution for constructed features.  However, I'm a field biologist.   I'm navigating around obstacles while mapping the periphery of a Habitat patch/polygon.  Say a stand of invasive plants.   And need to walk around a fallen tree, or deep hole in the water, or impenetrable brush, and would rather not have to turn the track on and off hundreds of times.   And the track has no idea if I'm actually following the feature of interest or not.

By the way,  I frequently am using a submeter Bluetooth receiver ( using mock location in Android) to get precise locations.   I can place the receiver exactly where I want and then record a poi/waypoint.  Move to the next location, and on and on.   And again,  a physical press  of a button would be a lot easier than a precise screen tap. Not a deal



Sent from my Verizon, Samsung Galaxy smartphone



-------- Original message --------
From: Om Prakash Jena ***@***.***>
Date: 1/13/24 5:55 PM (GMT-05:00)
To: androidseb/mapmarker ***@***.***>
Cc: StLarry7 ***@***.***>, Author ***@***.***>
Subject: Re: [androidseb/mapmarker] Remote control button for POI/waypoint (Issue #40)


Recording a track also a good feature.

**Comment by androidseb** [comment]
Thanks, yes that explanation makes sense, I see how track recording won't solve this use case, but if it helps a little, know that I'm currently working on tracks recording.

I wonder how your bluetooth headset setup works, does it require the device to keep the screen on?

It seems fairly easy to listen for volume key presses on Android if the view is active in the foreground, but then it looks much more challenging to detect those volume button presses when the app is in the background...

If you needed to keep the app open in the foreground with the screen on for this to work, would that be a problem?

**Comment by StLarry7** [comment]
No problem at all...the MapMarker app is in the foreground/active while I am mapping.
________________________________

**Comment by StLarry7** [comment]
I can't seem to find the volume key setting in the options.  Is this volume key configuration option live/active?  I updated to the most recent version, but can't seem to find the setting.
________________________________

**Comment by androidseb** [comment]
Hey, this issue is still in the backlog I haven't gotten to it yet. You'll see the status of this issue change when I pick it up.

**Comment by androidseb** [comment]
I have added this use case to the list of possibly related features under the Workflows feature: https://github.com/androidseb/mapmarker/issues/70

The "VOLUME UP" and "VOLUME DOWN" key presses on Android could be configured to start a custom Workflow. That, coupled the ability for a custom workflow to do either of these:

> * add a marker based on where the screen center aims at on the map
> * add a marker based on where the current device location is
> * start recording a new track

would address the use case of this feature idea.
